### PR TITLE
Refractor track to be more DRY

### PR DESCRIPTION
**Pull request by lherman-cs**
Resolves https://github.com/pion/mediadevices/issues/141

Before:
Screenshot from 2020-04-18 21-22-31

After:
Screenshot from 2020-04-18 21-22-25

### CODE REVIEW

File: sampler.go [hunk 1]
```diff
@@ -6,26 +6,29 @@ import (
...
 
-func (s *sampler) sample(b []byte) error {
-	now := time.Now()
-	duration := now.Sub(s.lastTimestamp).Seconds()
-	samples := uint32(s.clockRate * duration)
-	s.lastTimestamp = now
+	return samplerFunc(func(b []byte) error {
+		now := time.Now()
+		duration := now.Sub(lastTimestamp).Seconds()
+		samples := uint32(clockRate * duration)
```
Comment (at-wat) [inline sampler.go:20]: I think this should be rounded.

File: sampler.go [hunk 2]
```diff
@@ -6,26 +6,29 @@ import (
...
+		return t.WriteSample(media.Sample{Data: b, Samples: samples})
+	})
+}
 
-	sample := media.Sample{Data: b, Samples: samples}
-	return s.track.WriteSample(sample)
+// newAudioSampler creates a audio sampler that uses a fixed latency and
+// the codec's clock rate to come up with a duration for each sample.
+func newAudioSampler(t LocalTrack, latency time.Duration) samplerFunc {
+	samples := uint32(float64(t.Codec().ClockRate) * latency.Seconds())
```
Comment (at-wat) [inline sampler.go:30]: ditto

File: sampler.go (hunk 1 above)
Comment (lherman-cs) [reply sampler.go:20]: makes sense. I'll update it.

File: sampler.go (hunk 2 above)
Comment (lherman-cs) [reply sampler.go:30]: same

**Comment by lherman-cs** [comment]
@at-wat Updated the PR. PTAL.

**Pull request merged**